modules/tinymce/src/core/main/ts/content/TransparentElements.ts

```diff
@@ -0,0 +1,57 @@
+import { Arr, Obj, Type } from '@ephox/katamari';
+import { Compare, SugarElement, SugarNode, Traverse } from '@ephox/sugar';
+
+import AstNode from '../api/html/Node';
+import Schema, { SchemaMap } from '../api/html/Schema';
+import * as NodeType from '../dom/NodeType';
+
+export const transparentBlockAttr = 'data-mce-block';
+
+const makeSelectorFromSchemaMap = (map: SchemaMap) => Arr.filter(Obj.keys(map), (key) => /^[a-z]+$/.test(key)).join(',');
+
+const updateAt = (schema: Schema, root: Element, scope: Element, inEditorRoot: boolean) => {
+  const transparentSelector = makeSelectorFromSchemaMap(schema.getTransparentElements());
+  const blocksSelector = makeSelectorFromSchemaMap(schema.getBlockElements());
+
+  Arr.each(scope.querySelectorAll(transparentSelector), (transparent) => {
+    if ((inEditorRoot && transparent.parentElement === root) || transparent.querySelectorAll(blocksSelector).length > 0) {
+      transparent.setAttribute(transparentBlockAttr, 'true');
+
+      if (transparent.getAttribute('data-mce-selected') === 'inline-boundary') {
+        transparent.removeAttribute('data-mce-selected');
+      }
+    } else {
+      transparent.removeAttribute(transparentBlockAttr);
+    }
+  });
+};
+
+export const update = (schema: Schema, root: Element, inEditorRoot: boolean): void => updateAt(schema, root, root, inEditorRoot);
```

Comment:
```suggestion
export const update = (schema: Schema, root: Element, inEditorRoot: boolean): void =>
  updateAt(schema, root, root, inEditorRoot);
```
This is just a little easier to read 🤷 We've also mixed doing this below.

Reply:
I only move them down if they don't fit on a line. I think this is readable. Lambdas is supposed to be one liners if they can be.

Reply:
It makes sense when you don't have a lot of args, but here you do and just makes it many times harder to read.

Reply:
Not sure I agree anyway I could ETA reduce that function away it was just an alias now when the root check was removed.

Reply:
Doesn't matter then... anyways, to explain myself. When you need to search across a line to try and find the start it involves a lot more scanning. So at all times it's just easier to have the function definition and then the code separately. So as you add more to the definition it just gets harder to find the end of the definition and start of the actual code.

Code highlighting can help, but you don't always have access to that or some systems might not do it as well.

Reply:
If we only had a language that didn't have all these noisy type annotations.

Like I rather have this (rescript):
```
let filter = (v, ~f) => keep(v, f)
let filterWithIndex = (v, ~f) => keepWithIndex(v, f)
let filterMap = (v, ~f) => keepMap(v, f)
```

Than this it's so much harder to read for me at least:
```
let filter = (v, ~f) =>
  keep(v, f)
let filterWithIndex = (v, ~f) =>
  keepWithIndex(v, f)
let filterMap = (v, ~f) =>
  keepMap(v, f)
```

I think the => in typescript is easy to spot there is where the body starts and the annotations stop. Having less lines of code is also a plus. I would prefer to have just one or two pages of code per module very hard to do sometime when one just adds and adds things and want to hide the details of a module.

We can debate this in slack if you like. I don't think there is a single truth here.